src/metabase/lib/util.cljc

```diff
@@ -475,6 +475,16 @@
   [query :- :map]
   (= (first-stage-type query) :mbql.stage/native))
 
+(def ^:dynamic ^{:arglists '(^java.lang.String [^java.lang.String s])} *escape-alias-fn*
```

Comment:
When using MLv2 to determine what column names will appear in a query (in order to generate nominal field literal refs) we need to make sure we're generating unique aliases the same way we do in the old `add-alias-info` namespace, otherwise things will break. Added a dynamic variable that gets bound to `(fn [s] (driver/escape-alias driver/*driver* s))` automatically when this is used inside the QP